// Copyright 2020 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe.autoflip;

import "mediapipe/framework/calculator.proto";

message ShotBoundaryDecoderCalculatorOptions {
  extend mediapipe.CalculatorOptions {
    optional ShotBoundaryDecoderCalculatorOptions ext = 275222224;
  }

  // The threshold to determine whether a frame is the end of a scene.
  // If the decoded output of the TransVetV2 is greater than it, the 
  // corresponding frame is the end of a scene. Otherwise, the corresponding
  // frame is not the end of a scene. 
  optional double threshold = 1 [default = 0.5];

  // Minimum number of shot duration (in seconds).
  optional double min_shot_span = 2 [default = 0];

  // Only send results if the shot value is true.
  optional bool output_only_on_change = 3 [default = true];

}
